python/foglamp/plugins/south/sinusoid/sinusoid.py

```diff
@@ -49,6 +48,111 @@
 _LOGGER = logger.setup(__name__, level=logging.INFO)
 index = -1
 _task = None
+loop = None
+c_callback = None
+c_ingest_ref = None
```

Comment:
Can we keep these 3 variables in handle or some BasePlugin Object? so that we don't need to make these global.